airflow/utils/log/file_task_handler.py

```diff
@@ -147,7 +147,7 @@ class FileTaskHandler(logging.Handler):
 
     def __init__(self, base_log_folder: str, filename_template: str | None = None):
         super().__init__()
-        self.handler: logging.FileHandler | None = None
+        self.handler: logging.Handler | None = None
```

Comment:
This change look out of place?

Reply:
This is true - but it allows the lower level handler, `_RedisHandler` to not inherit from `logging.FileHandler`, which I think is appropriate since there is no file involved(?)

It looks like the codebase already doesn't enforce that the lower level handler extends from `logging.FileHandler` - https://github.com/apache/airflow/blob/2940b9fa55a6a72c60c2162e541631addec3d6b8/airflow/providers/amazon/aws/log/cloudwatch_task_handler.py#L67 which is defined in https://github.com/kislyuk/watchtower/blob/698ff2241befb6586269b6bbce0e305032c2d45c/watchtower/__init__.py#L139

This touches on the bits that I'm unsure about - it looks like `FileTaskHandler` has a lot of file-related things, but also it looks like it's expected to extend from it when not logging to a file. Not quite sure how to tackle that

Reply:
Should have been more specific initially. :)

What I meant was that maybe this change is not required by this PR? since you have anyway overridden the         `self.handler` in class RedisTaskHandler. Also, `FileTaskHandler` is used in many places, and changing it here might affect those implementations. 

About inheriting from `FileTaskHandler` I'm not sure either, maybe @bolkedebruin might have more context.

Reply:
> What I meant was that maybe this change is not required by this PR? since you have anyway overridden the self.handler in class RedisTaskHandler

Ah - so if I change it back to how it was before this PR, I get a type error:
```
airflow/providers/redis/log/redis_task_handler.py:61: error: Incompatible types
in assignment (expression has type "Optional[_RedisHandler]", base class
"FileTaskHandler" defined the type as "Optional[FileHandler]")  [assignment]
```
So something else would have to change too. The easiest way that I've come up with is to de-facto disable type checking on `self.handler` by removing type annotations in a few places, to make it more like the Cloudwatch handler, but that seems a bit... cheat-y.

`_RedisHandler` could really inherit from `logging.FileHandler` to satisfy type checking, but it's constructor seems to require a path to a file, which seems odd in this case.

Reply:
This change is OK to me since the base class only really needs to inner handler to be a `logging.Handler`. It could be moved to a separate PR but since this is where the change becomes necessary (instead of just a generally good thing to do) it’s fine to be included here.

Reply:
If this specific change is not tightly coupled to this PR I would suggest to do it in a separated PR.
this is because utils/log goes to core release cycle while the other files goes to provider release cycle.
@michalc Is it possible to seperate?

Reply:
@michalc can you check my above comment?

Reply:
@eladkal Sounds reasonable to separate to another PR. Will do that now

Reply:
@eladkal Done in https://github.com/apache/airflow/pull/32773